src/text_info.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use std::sync::Arc;
+use text_lines::TextLines;
+
+use super::types::LineAndColumnDisplay;
+use super::types::LineAndColumnIndex;
+use crate::swc::common::BytePos;
+use crate::swc::common::Span;
+
+#[derive(Clone)]
+pub struct SourceTextInfo {
+  start_pos: BytePos,
+  text: Arc<String>,
+  text_lines: Arc<TextLines>,
+}
+
+impl SourceTextInfo {
+  pub fn new(start_pos: BytePos, text: Arc<String>) -> Self {
+    // the BOM should ideally be stripped before being
+    // provided to this struct
+    debug_assert!(!text.starts_with(BOM_CHAR));
```

Comment:
What happens if not in debug and the byte-order-mark hasn't been stripped?

Reply:
I updated the comment to explain that. In summary, it should be no big deal, but it would be better if people don't provide it because that's a text encoding concern that this module should ideally not have to worry about.

Reply:
Nevermind. It failed on the BOM test in CLI. I've updated this to strip the BOM, but we should ideally not be doing that here (it's a minor point) and so I've added a todo.